client/src/mid.rs

```diff
@@ -449,166 +896,385 @@ mod tests {
-                        ..Default::default()
-                    }),
-                    Ok(ReadTaskShortResponse {
-                        task_id: 1,
-                        name: "Test Task 2".into(),
-                        ..Default::default()
-                    }),
-                    Err("random error message".into()),
-                    Ok(ReadTaskShortResponse {
-                        task_id: 2,
-                        name: "Test Task 3".into(),
-                        ..Default::default()
-                    }),
-                ])
-                .unwrap(),
-            )
-            .expect(1)
-            .create_async()
-            .await;
-
-        server
-            .mock("GET", mockito::Matcher::Any)
-            .with_body("TEST MAIN PATH")
-            .expect(0)
-            .create_async()
-            .await;
-
+    // tests the State init function, also used to init tests
+    async fn test_init() -> (ServerGuard, State, Receiver<MidEvent>, ViewKey) {
+        let server = mockito_setup().await;
         let url = server.url();
         println!("url: {url}");
 
         // init state
-        let state = init(&url).await.unwrap();
+        let (mut state, mut receiver) = init(&url).unwrap();
+        // await server response for FilterRequest
+        state.handle_mid_event(receiver.next().await.unwrap());
+        println!("ui event {:?}", receiver.next().await.unwrap()); // drop UI event
+                                                                   // // await server response for ReadTasksShortResponse (request automatically sent when handle_mid_event is called on FilterResponse)
+                                                                   //dbg!(receiver.next().await.unwrap());
+        state.handle_mid_event(receiver.next().await.unwrap());
+        println!("ui event {:?}", receiver.next().await.unwrap()); // drop UI event
 
         // make sure view was created with correct state
-        let view = state.view_get(state.view_get_default().unwrap()).unwrap();
+        let view_key = state.view_get_default().unwrap();
+        let view = state.view_get(view_key).unwrap();
         let mut i = 0;
         view.tasks.as_ref().unwrap().iter().for_each(|t| {
             assert_eq!(state.task_get(*t).unwrap().db_id.unwrap(), i);
             i += 1;
         });
+
+        (server, state, receiver, view_key)
     }
 
-    #[test]
-    fn test_frontend_api() {
-        // test view_def, view_mod & task_def
-        let mut state = init_test();
-        dbg!(&state);
-        let view_key = state.view_get_default().unwrap();
-        // test view_get
+    /// get event with timeout from midevent receiver
+    pub async fn get_event(receiver: &mut Receiver<MidEvent>) -> MidEvent {
+        timeout(Duration::from_millis(100), receiver.next())
+            .await
+            .unwrap()
+            .unwrap()
+    }
+
+    #[tokio::test]
+    // #[traced_test]
+    async fn test_tasks() {
```

Comment:
what was the correction that fixed this test? I remember it being a problem earleir this week?